compiler/il/AliasSetInterface.hpp

```diff
@@ -36,19 +36,31 @@
 #include "infra/Assert.hpp"
 #include "infra/BitVector.hpp"
 
+typedef enum {
+   useDefAliasSet,
+   UseOnlyAliasSet
+} AliasSetInterface;
 
-template <class AliasSetInterface>
+template <uint32_t _AliasSetInterface>
 class TR_AliasSetInterface {
 public:
+   
+   TR_AliasSetInterface(TR::SymbolReference *symRef, bool isDirectCall = false, bool includeGCSafePoint = false) :
+      _isDirectCall(isDirectCall),
+      _includeGCSafePoint(includeGCSafePoint),
+      _symbolReference(symRef),
+      _shares_symbol(true) {}
 
-   TR_AliasSetInterface(bool isDirectCall = false, bool includeGCSafePoint = false) :
+   TR_AliasSetInterface(bool shares_symbol, TR::SymbolReference *symRef, bool isDirectCall = false, bool includeGCSafePoint = false) :
       _isDirectCall(isDirectCall),
-      _includeGCSafePoint(includeGCSafePoint)
-      {}
+      _includeGCSafePoint(includeGCSafePoint),
+      _symbolReference(symRef),
+      _shares_symbol(shares_symbol) {}
+
 
    TR_BitVector *getTRAliases()
       {
-      return static_cast<AliasSetInterface*>(this)->getTRAliases_impl(_isDirectCall, _includeGCSafePoint);
+      return (this)->getTRAliases_impl(_isDirectCall, _includeGCSafePoint);
```

Comment:
do we even need the `(this)->` - seems redundant...

Reply:
Remove the `(this)->` in 091c9f0